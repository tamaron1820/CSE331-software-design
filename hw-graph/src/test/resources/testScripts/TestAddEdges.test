# Tests that add edges correctly
# Test if same edge is added graph should not change
# Test if add another edge between two nodes which are already connected by
# different edge, the edge should be added

# Create a graph
CreateGraph graph1

# Add nodes
AddNode graph1 n1
AddNode graph1 n2

# Add edges
AddEdge graph1 n1 n2 e1
AddEdge graph1 n1 n2 e2
AddEdge graph1 n1 n2 e1
AddEdge graph1 n2 n1 e3

# Prints the nodes in the graph and the outgoing edges from n1 and n2
ListNodes graph1
ListChildren graph1 n1
ListChildren graph1 n2