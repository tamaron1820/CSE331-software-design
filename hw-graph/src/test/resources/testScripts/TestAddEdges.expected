# Tests that add edges correctly
# Test if same edge is added graph should not change
# Test if add another edge between two nodes which are already connected by
# different edge, the edge should be added

# Create a graph
created graph graph1

# Add nodes
added node n1 to graph1
added node n2 to graph1
added node n3 to graph1

# Add an edge
added edge e1 from n1 to n2 in graph1
added edge e2 from n1 to n2 in graph1
added edge e1 from n1 to n2 in graph1
added edge e3 from n3 to n1 in graph1
added edge e4 from n2 to n1 in graph1

# Print the nodes in the graph and the outgoing edges from n1 and n2,n3
graph1 contains: n1 n2 n3
the children of n1 in graph1 are: n2(e1) n2(e2)
the children of n2 in graph1 is: n1(e4)
the children of n3 in graph1 is: n1(e3)